activation/handler.go

```diff
@@ -165,93 +125,84 @@ func (h *Handler) SyntacticallyValidate(ctx context.Context, atx *types.Activati
 		if atx.InitialPost == nil {
 			return errors.New("no prev atx declared, but initial post is not included")
 		}
-		if atx.InnerActivationTx.NodeID == nil {
+		if atx.NodeID == nil {
 			return errors.New("no prev atx declared, but node id is missing")
 		}
 		if atx.VRFNonce == nil {
 			return errors.New("no prev atx declared, but vrf nonce is missing")
 		}
-		if atx.CommitmentATX == nil {
+		if atx.CommitmentATXID == nil {
 			return errors.New("no prev atx declared, but commitment atx is missing")
 		}
-		if *atx.CommitmentATX == types.EmptyATXID {
+		if *atx.CommitmentATXID == types.EmptyATXID {
 			return errors.New("empty commitment atx")
 		}
 		if atx.Sequence != 0 {
 			return errors.New("no prev atx declared, but sequence number not zero")
 		}
 
-		// Use the NIPost's Post metadata, while overriding the challenge to a zero challenge,
-		// as expected from the initial Post.
-		initialPostMetadata := *atx.NIPost.PostMetadata
-		initialPostMetadata.Challenge = shared.ZeroChallenge
 		if err := h.nipostValidator.VRFNonce(
-			atx.SmesherID, *atx.CommitmentATX, atx.VRFNonce, &initialPostMetadata, atx.NumUnits,
+			atx.SmesherID, *atx.CommitmentATXID, *atx.VRFNonce, atx.NumUnits,
 		); err != nil {
 			return fmt.Errorf("invalid vrf nonce: %w", err)
 		}
+		post := wire.PostFromWireV1(atx.InitialPost)
 		if err := h.nipostValidator.Post(
-			ctx, atx.SmesherID, *atx.CommitmentATX, atx.InitialPost, &initialPostMetadata, atx.NumUnits,
+			ctx, atx.SmesherID, *atx.CommitmentATXID, post, shared.ZeroChallenge, atx.NumUnits,
 		); err != nil {
 			return fmt.Errorf("invalid initial post: %w", err)
 		}
 	default:
-		if atx.InnerActivationTx.NodeID != nil {
+		if atx.NodeID != nil {
 			return errors.New("prev atx declared, but node id is included")
 		}
 		if atx.InitialPost != nil {
 			return errors.New("prev atx declared, but initial post is included")
 		}
-		if atx.CommitmentATX != nil {
+		if atx.CommitmentATXID != nil {
 			return errors.New("prev atx declared, but commitment atx is included")
 		}
 	}
 	return nil
 }
 
-func (h *Handler) SyntacticallyValidateDeps(
+func (h *Handler) syntacticallyValidateDeps(
 	ctx context.Context,
-	atx *types.ActivationTx,
-) (*types.VerifiedActivationTx, *mwire.MalfeasanceProof, error) {
-	var (
-		commitmentATX *types.ATXID
-		err           error
-	)
+	atx *wire.ActivationTxV1,
+) (leaves uint64, effectiveNumUnits uint32, proof *mwire.MalfeasanceProof, err error) {
+	var commitmentATX types.ATXID
+
 	if atx.PrevATXID == types.EmptyATXID {
-		if err := h.validateInitialAtx(ctx, atx); err != nil {
-			return nil, nil, err
+		if err := h.nipostValidator.InitialNIPostChallenge(&atx.NIPostChallengeV1, h.cdb, h.goldenATXID); err != nil {
```

Comment:
I believe usually we pass the db as first parameter (if there is no context):

```suggestion
		if err := h.nipostValidator.InitialNIPostChallenge(h.cdb, &atx.NIPostChallengeV1, h.goldenATXID); err != nil {
```

Reply:
IDK, there are many places in the code where a DB is not the first (after optional context) parameter. If you want to make it consistent, how about creating a dedicated PR to change the order of arguments through the codebase to have db first?